scripts/report/_01_make_reports.py

```diff
@@ -1514,6 +1519,159 @@ def add_decoding_grand_average(
             plt.close(fig)
 
 
+def add_csp_grand_average(
+    *,
+    session: str,
+    cfg: SimpleNamespace,
+    report: mne.Report,
+):
+    """Add CSP decoding results to the grand average report."""
+    import matplotlib.pyplot as plt  # nested import to help joblib
+
+    bids_path = BIDSPath(
+        subject='average',
+        session=session,
+        task=cfg.task,
+        acquisition=cfg.acq,
+        run=None,
+        recording=cfg.rec,
+        space=cfg.space,
+        suffix='decoding',
+        extension='.mat',
+        datatype=cfg.datatype,
+        root=cfg.deriv_root,
+        check=False
+    )
+
+    for contrast in cfg.decoding_contrasts:
+        cond_1, cond_2 = contrast
+        a_vs_b = f'{cond_1}+{cond_2}'.replace(op.sep, '')
+        processing = f'{a_vs_b}+CSP+{cfg.decoding_metric}'
+        processing = processing.replace('_', '-').replace('-', '')
+        fname_csp_cluster_results = bids_path.copy().update(
+            processing=processing,
+        )
+        csp_cluster_results = loadmat(fname_csp_cluster_results)
+
+        for freq_range_name in cfg.decoding_csp_freqs.keys():
+            results = csp_cluster_results[freq_range_name][0][0]
+            mean_crossval_scores = results['mean_crossval_scores']
+            # t_vals = results['t_vals']
+            clusters = results['clusters']
+            cluster_p_vals = results['cluster_p_vals'].squeeze()
+            time_bin_edges = results['time_bin_edges'].squeeze()
+            freq_bin_edges = results['freq_bin_edges'].squeeze()
+            cluster_t_threshold = results['cluster_t_threshold'].item()
+
+            significant_cluster_idx = np.where(
+                cluster_p_vals < cfg.cluster_permutation_p_threshold
+            )[0]
+            significant_clusters = clusters[significant_cluster_idx]
+
+            # vmax = np.abs(t_vals).max()
+            # vmin = -vmax
+            vmax = max(
+                np.abs(mean_crossval_scores.min() - 0.5),
+                np.abs(mean_crossval_scores.max() - 0.5)
+            ) + 0.5
+            vmin = 0.5 - (vmax - 0.5)
```

Comment:
this only makes sense for roc_auc, balanced accuracy or accuracy if data have been balanced before training.

Reply:
I've added an `assert` here to ensure we're only working with `roc_auc` for now.

There are other places in the pipeline where we silently assume a chance level of 50%, hence we should probably address this entire issue in a separate PR